src/main/java/org/assertj/core/internal/Maps.java

```diff
@@ -572,7 +570,7 @@ public void assertHasSameSizeAs(AssertionInfo info, Map<?, ?> map, Map<?, ?> oth
    * @throws AssertionError if the actual map not contains the given key.
    */
   public <K, V> void assertContainsKeys(AssertionInfo info, Map<K, V> actual,
-                                        @SuppressWarnings("unchecked") K... keys) {
+                                        K... keys) {
```

Comment:
surprisingly eclipse does complain about the head pollution when I remove `@SuppressWarnings("unchecked")`

Reply:
Same comment as for the `Classes`. It seems that the ecj compiler is slightly different that the javac in this area.